### DISCUSSION TITLE
Change the default log color

### DISCUSSION THREAD

**Discussion by squatto** [spatie/ray#330] [Ideas]
Currently you have to set the color on each `ray()` call, which can get rather tedious when you have a larger number of calls:

```php
ray('log this')->red();
ray('and log this as well')->red();
ray('user', $user)->red();
ray($results)->red();
ray($anotherVariable)->red();
// etc...
```

It would be useful to be able to set the default color to be used for log calls that don't explicitly define a color. This would make it easy to do things like setting a color at the start of a method and resetting the color at the end of the method. It would then be easy to identify the log lines triggered by that method (and everything that is called from within that method).

Something along the lines of `ray()->setDefaultColor()->red()` (or any of the color methods). You could then do something like `ray()->resetDefaultColor()` to reset the default color.

Here's a simple usage example:

```
ray('No color');

ray()->setDefaultColor()->red();

ray('This will be red');

ray('This will be blue because it is explicitly set')->blue();

ray('This will be red because it is still set as the default color');

ray()->resetDefaultColor();

ray('This will not have a color');
```

I'd love to know your thoughts. Thanks!

**Comment by freekmurze** [comment]
Thanks for your idea

Not too many people have asked for this, so I'm probably going to leave it as it is for now, but I'll reconsider should more people as for this.

**Reply by squatto** [reply]
Sounds great, thanks for considering it!

For some context, I was writing a Ray macro `cycleColor()` that would cycle between the colors. I was going to use it on each iteration of a loop to easily distinguish log messages from each iteration. Then I realized that I couldn't change the default color and had to pass it to each `ray()` call, which led to submitting this idea.

Thanks again!
